lib/modules/platform/bitbucket/index.ts

```diff
@@ -113,17 +114,30 @@ export async function initPlatform({
 }
 
 // Get all repositories that the user has access to
-export async function getRepos(): Promise<string[]> {
+export async function getRepos(config?: AutodiscoverConfig): Promise<string[]> {
   logger.debug('Autodiscovering Bitbucket Cloud repositories');
   try {
-    const repos = (
+    let repos = (
       await bitbucketHttp.getJson<PagedResult<RepoInfoBody>>(
         `/2.0/repositories/?role=contributor`,
         {
           paginate: true,
         },
       )
     ).body.values;
+
+    // if autodiscoverProjects is configured
+    // filter the repos list
+    if (config?.projects?.length) {
+      logger.debug(
+        { autodiscoverProjects: config?.projects },
+        'Applying autodiscoverProjects filter',
+      );
+      repos = repos.filter((repo) =>
+        matchRegexOrGlobList(repo.project.name, config.projects!),
```

Comment:
```suggestion
        matchRegexOrGlobList(repo.project.name, config.projects),
```
you already validate above

Reply:
TS throws errors for some reason :)